runtime/error.c

```diff
@@ -1,9 +1,138 @@
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+
 #include "balrt.h"
+#include "third-party/libbacktrace/backtrace.h"
+
+#define INITIAL_FRAME_COUNT 1
+#define MAX_FRAME_COUNT 200
+#define MAX_BYTES_COUNT INT32_MAX
+#define INITIAL_BYTES_COUNT 8
+#define SKIP_FROM_END 1 // This number counts from the last stack frame
+#define THREAD 0
+
+typedef struct {
+    uint32_t nFrames;
+    uint32_t cFrames;
+    uint32_t nBytes;
+    uint32_t cBytes;
+    Frame *frames;
+    char *bytes;
+} FramesInfo;
```

Comment:
Isn't this basically the `Trace` struct?
(Minor point: personally don't like the word `Info`.)

Reply:
(I'm not keen on `Info` either, although I have been guilty of using it.)

`cBytes` is not comprehensible: `size` (which abbreviates to `sz`) is another word you can use instead of capacity.

Reply:
Changed